servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Single.java

```diff
@@ -64,6 +64,76 @@
  * @param <T> Type of the result of the single.
  */
 public abstract class Single<T> {
+
+    /**
+     * Callback interface on which only a single method is ever called matching the terminal outcome of the associated
+     * {@link Single} and {@link Cancellable}.
+     *
+     * @param <T> Type of the result of the {@link Single}.
+     */
+    public interface TerminalSignalConsumer<T> {
```

Comment:
consider moving this new code down in the class as this interface isn't a central feature of the outer class.